### PR TITLE
Feature: Further ssh script simplify

### PR DESCRIPTION
**Pull request by h-alice**
Implemented the SSHD config generator into `views.py`.
Also somewhere refactored the code to make it cleaner.

### CODE REVIEW

File: src/tunnels/tunnel_script_renderer/templates/ssh_config_template.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Host ${host_friendly_name}
+    HostName ${server_domain}
+    Port ${reverse_port} ${reverse_server_ssh_port}
+    Compression yes
```
Comment (NatLee) [inline src/tunnels/tunnel_script_renderer/templates/ssh_config_template.conf:4]: Will there be any loss of performance in the connection between SSH servers if we set `Compression`?

File: src/tunnels/views.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import re
 from enum import Enum
+from tkinter import SE
```
Comment (NatLee) [inline src/tunnels/views.py:3]: Wow! This is my first time to meet this Class. :O

File: src/tunnels/views.py [hunk 3]
```diff
@@ -12,6 +13,30 @@
...
 from authorized_keys.models import ReverseServerAuthorizedKeys
 
+from authorized_keys.utils import ssh
+from tunnel_script_renderer import ssh_tunnel_script_factory
+from tunnel_script_renderer import sshd_client_config_factory, sshd_server_config_factory
+
+## Static constants
+CLIENT_STEM = """
+# ========================================
+# Endpoint Configuration
```
Comment (NatLee) [inline src/tunnels/views.py:23]: I am uncertain about this naming. 🤔
I do not believe that `Endpoint` is appropriate in my original code.
Would replacing it with `Tunnel` be more suitable?

File: src/tunnels/views.py [hunk 4]
```diff
@@ -236,68 +242,11 @@ def get_script(
...
-        Start-Sleep -Seconds 5
-    }}
-}} finally {{
-    Prevent-Sleep -Enable $false
-    Write-TimestampedMessage "Sleep prevention disabled."
-    echo "[+] Script end"
-    Write-Host "Press any key to continue..."
-    $Host.UI.RawUI.ReadKey("NoEcho,IncludeKeyDown")
-}}"""
+        config_string = ssh_tunnel_script_factory("powershell",
```
Comment (NatLee) [inline src/tunnels/views.py:245]: This appears to be related to cleaning. Thank you for your contribution. 🤩🤩🤩

File: src/tunnels/views.py (hunk 2 above)
Comment (NatLee) [reply src/tunnels/views.py:3]: It appears that there may be a typo with autocomplete. 🤣

**Review by NatLee — APPROVED** [review]
This commit appears to be acceptable and is ready to be merged with the main branch. Thank you for this template. 🥳

**Comment by NatLee** [comment]
@h-alice sorry, it shall not pass at this time. QwQ

image

**Review by NatLee — APPROVED** [review]
It's OK now. uwu

File: src/tunnels/tunnel_script_renderer/ssh_config_renderer.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+# NOTE: change if the directory structure changed.
+TEMPLATE_DIR = Path(__file__).resolve().parent / Path("templates")
+
+SSH_CLIENT_PROXY_CONF = "ProxyCommand ssh -W %h:%p telepy-ssh-server"
+
+class BaseSshTemplate(BaseTemplateRenderer):
+
+    def __init__(self, 
+                 server_side: bool, 
+                 host_friendly_name: str,
```
[the remark this answers is not in the document]
Comment (h-alice) [reply src/tunnels/tunnel_script_renderer/ssh_config_renderer.py:17]: Friendly : D

File: src/tunnels/tests/test_ssh_full_renderer.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import unittest
+from unittest.mock import Base
+from tunnel_script_renderer import sshd_client_config_factory, sshd_server_config_factory
+from tunnel_script_renderer import SshClientTemplate, SshServerTemplate
+
+
+class TestSshClientTemplate(unittest.TestCase):
+    def setUp(self):
+        self.template_client = sshd_client_config_factory("halice-server", "test_user", 1234)
```
[the remark this answers is not in the document]
Comment (h-alice) [reply src/tunnels/tests/test_ssh_full_renderer.py:9]: uwu

File: src/tunnels/tunnel_script_renderer/__init__.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+from .template_renderer import BaseTemplateRenderer
+from .powershell_renderer import PowerShellTemplate
+from .autossh_renderer import AutoSshTemplate
+from .ssh_config_renderer import SshClientTemplate, SshServerTemplate
+
+
+def ssh_tunnel_script_factory(tunnel_type: str, server_domain: str, reverse_port: int, ssh_port: int, reverse_server_ssh_port: int) -> BaseTemplateRenderer:
+    
+    tunnel_type = tunnel_type.lower() # Normalize `tunnel_type`.
+    if tunnel_type == "powershell":
```
[the remark this answers is not in the document]
Comment (h-alice) [reply src/tunnels/tunnel_script_renderer/__init__.py:10]: You're right, we can do a quick fix about it.

File: src/tunnels/views.py (hunk 4 above)
Comment (h-alice) [reply src/tunnels/views.py:247]: : D

File: src/tunnels/views.py (hunk 2 above)
Comment (h-alice) [reply src/tunnels/views.py]: Where did it come from ??? 🤣
This is a GUI interface for Python, and it does not related to our project.

File: src/tunnels/views.py (hunk 3 above)
Comment (h-alice) [reply src/tunnels/views.py:21]: It's OK : D

**Comment by h-alice** [comment]
What the hell is tkinter????
WHYYYY

**Pull request merged**